tiledb/api/src/group/mod.rs

```diff
@@ -0,0 +1,556 @@
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+        T: AsRef<str>,
+    {
+        let ctx = context.capi();
+        let c_uri = cstring!(uri.as_ref());
+        let c_ptr = match name {
+            None => std::ptr::null::<std::ffi::c_char>(),
+            Some(s) => cstring!(s.as_ref()).as_ptr(),
+        };
+        let c_relative: u8 = if relative { 1 } else { 0 };
+        context.capi_return(unsafe {
+            ffi::tiledb_group_add_member(
+                ctx,
+                Self::capi(self),
+                c_uri.as_ptr(),
+                c_relative,
+                c_ptr,
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn delete_member<S>(
+        &self,
+        context: &'ctx Context,
+        name_or_uri: S,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let ctx = context.capi();
+        let c_name_or_uri = cstring!(name_or_uri.as_ref());
+        context.capi_return(unsafe {
+            ffi::tiledb_group_remove_member(
+                ctx,
+                Self::capi(self),
+                c_name_or_uri.as_ptr(),
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn get_member_count(
+        &self,
+        context: &'ctx Context,
+    ) -> TileDBResult<u64> {
+        let ctx = context.capi();
+        let mut c_count: u64 = out_ptr!();
+        context.capi_return(unsafe {
+            ffi::tiledb_group_get_member_count(
+                ctx,
+                Self::capi(self),
+                &mut c_count,
+            )
+        })?;
+        Ok(c_count)
+    }
+
+    pub fn get_member_by_index(
+        &self,
+        context: &'ctx Context,
+        index: u64,
+    ) -> TileDBResult<GroupInfo> {
+        let ctx = context.capi();
+        let mut tiledb_uri = std::ptr::null_mut::<ffi::tiledb_string_t>();
+        let mut tiledb_name = std::ptr::null_mut::<ffi::tiledb_string_t>();
+        let mut tiledb_type: ffi::tiledb_object_t = out_ptr!();
+        context.capi_return(unsafe {
+            ffi::tiledb_group_get_member_by_index_v2(
+                ctx,
+                Self::capi(self),
+                index,
+                &mut tiledb_uri as *mut *mut ffi::tiledb_string_t,
+                &mut tiledb_type,
+                &mut tiledb_name as *mut *mut ffi::tiledb_string_t,
+            )
+        })?;
+        let uri = TDBString {
+            raw: RawTDBString::Owned(tiledb_uri),
+        }
+        .to_string()?;
+
+        let name = TDBString {
+            raw: RawTDBString::Owned(tiledb_name),
+        }
+        .to_string()?;
+
+        let object_type = ObjectType::try_from(tiledb_type)?;
+        Ok(GroupInfo {
+            uri,
+            group_type: object_type,
+            name,
+        })
+    }
+
+    pub fn get_member_by_name<S>(
+        &self,
+        context: &'ctx Context,
+        name: S,
+    ) -> TileDBResult<GroupInfo>
+    where
+        S: AsRef<str>,
+    {
+        let ctx = context.capi();
```

Comment:
Most places this is named `c_context`, I think it's a good convention to follow since the name tells us immediately that it's something we will pass to an `ffi` function.